File: C-Otoshidama.go
package main
import "fmt"
import "os"

func main(){
	var n, y int
	fmt.Scan(&n, &y)
	for i := 0; i < n + 1; i ++ {
		for j := 0; j < n - i + 1; j ++ {
			if 10000 * i + 5000 * j == y - 1000 * (n - i - j) {
				fmt.Println(i, j, (n - i - j))
				os.Exit(0)
			}
		}
	}
	fmt.Println(-1, -1, -1)
}

File: C-Otoshidama.py
N, Y = map(int,input().split())

for i in range(N + 1):
    S = 0
    a = 10000 * i
    for j in range(N + 1):
        b = 5000 * j
        k = N - i - j
        if (Y - a - b) == 1000 * k:
            if k >= 0:
                print(i, j, k)
            
                exit()        

print('-1 -1 -1')
